packages/terra-tabs/src/_CollapsibleTabs.jsx

```diff
@@ -165,60 +163,25 @@ class CollapsibleTabs extends React.Component {
-
-    if (tabList || tabMoreBtn) {
-      const isRTL = document.getElementsByTagName('html')[0].getAttribute('dir') === 'rtl';
-      const visibleChildren = this.container.children;
-
-      if (event.nativeEvent.keyCode === KeyCode.KEY_LEFT) {
-        if (isRTL) {
-          this.handleFocusRight(visibleChildren, event);
-        } else {
-          this.handleFocusLeft(visibleChildren, event);
-        }
-      } else if (event.nativeEvent.keyCode === KeyCode.KEY_RIGHT) {
-        if (isRTL) {
-          this.handleFocusLeft(visibleChildren, event);
-        } else {
-          this.handleFocusRight(visibleChildren, event);
-        }
+    const tabList = event.target.getAttribute('role') === 'tab';
+    const isRTL = document.getElementsByTagName('html')[0].getAttribute('dir') === 'rtl';
+    const nextKey = !isRTL ? KeyCode.KEY_RIGHT : KeyCode.KEY_LEFT;
+    const previousKey = !isRTL ? KeyCode.KEY_LEFT : KeyCode.KEY_RIGHT;
+    let tabPanes = this.container.querySelectorAll('[role="tab"]');
+    tabPanes = (Array.from(tabPanes).filter((pane) => pane.hasAttribute('data-terra-tabs-show-focus-styles') && pane.getAttribute('data-terra-tabs-show-focus-styles') === 'true'));
+    const index = tabPanes.indexOf(event.target);
+    if (tabList) {
+      if ((event.nativeEvent.keyCode === KeyCode.KEY_RETURN || event.nativeEvent.keyCode === KeyCode.KEY_SPACE) && !event.target.getAttribute('data-terra-tabs-menu')) {
+        this.props.onChange(event, this.props.children[event.target.getAttribute('index')]);
       }
     }
-  }
-
-  handleFocusRight(visibleChildren, event) {
-    if (this.props.activeIndex >= this.hiddenStartIndex) {
-      return;
-    }
-
-    for (let i = this.props.activeIndex + 1; i < visibleChildren.length; i += 1) {
-      if (!this.props.children[i].props.isDisabled) {
-        if (visibleChildren[i] === this.menuRef) {
-          this.menuRef.focus();
-          break;
-        } else {
-          this.props.onChange(event, this.props.children[i]);
-          break;
-        }
-      }
-    }
-  }
-
-  handleFocusLeft(visibleChildren, event) {
-    let startIndex = this.props.activeIndex - 1;
-    if (startIndex >= this.hiddenStartIndex || document.activeElement === this.menuRef) {
-      startIndex = this.hiddenStartIndex - 1;
-    }
-
-    for (let i = startIndex; i >= 0; i -= 1) {
-      if (!this.props.children[i].props.isDisabled) {
-        if (document.activeElement === this.menuRef) {
-          this.container.focus();
-        }
-        this.props.onChange(event, this.props.children[i]);
-        break;
+    if (event.nativeEvent.keyCode === nextKey && tabPanes[index + 1]) {
+      tabPanes[index + 1].focus();
+    } else if (event.nativeEvent.keyCode === previousKey) {
+      if (tabPanes[index - 1]) {
+        tabPanes[index - 1].focus();
+      } else if (event.target === tabPanes[tabPanes.length - 1] && tabPanes[tabPanes.length - 2]) {
+        tabPanes[tabPanes.length - 2].focus();
```

Comment:
why do we need [length - 2] ?? is it added for focusing collapsed tabs

Reply:
^^

Reply:
Removed